### PR TITLE
Allow to use env file in image-builder when buildInADO is true

### PR DESCRIPTION
**Pull request by Sawthis**
**Description**

Changes proposed in this pull request:

- Allow to use env file in image-builder when buildInADO is true

**Related issue(s)**

https://github.com/kyma-project/test-infra/issues/10413

### CODE REVIEW

**Review by IwonaLanger — DISMISSED** [review]

File: cmd/image-builder/README.md [hunk 1]
```diff
@@ -197,7 +197,7 @@ The BuildKit and kaniko backends are deprecated and will be removed in the futur
 ### Azure DevOps Backend (ADO)
 
 The ADO backend uses Image Builder to call ADO API and trigger the `oci-image-builder` pipeline. This backend is SLC-29 compliant. It supports signing images with a production signify service. Images built with ADO can be pushed into Kyma Google Cloud artifacts registries. To build images, the ADO backend uses the `kaniko-project/executor` image. 
-This backend doesn't support the `--env-file`, `--platform`, and `--variant` flags. Building images for platforms other than amd64 is not supported. 
+This backend doesn't support the `--platform`, and `--variant` flags. Building images for platforms other than amd64 is not supported.
```
Comment (IwonaLanger) [inline cmd/image-builder/README.md:200]:
```suggestion
This backend doesn't support the `--platform` and `--variant` flags. Building images for platforms other than amd64 is not supported. 
```

File: cmd/image-builder/main.go [hunk 2]
```diff
@@ -477,6 +471,15 @@ func appendMissing(target *map[string]string, source []tags.Tag) {
 	}
 }
 
+// appendMissing appends key, values pairs from source array to target map
```
Comment (akiioto) [inline cmd/image-builder/main.go:474]: wrong name function

Comment (akiioto) [inline cmd/image-builder/main.go:474]:
```suggestion
// appendToTags appends key-value pairs from source map to target slice and returns the result
```

File: cmd/image-builder/main.go [hunk 3]
```diff
@@ -477,6 +471,15 @@ func appendMissing(target *map[string]string, source []tags.Tag) {
...
+func appendToTags(target *[]tags.Tag, source map[string]string) {
+	if len(source) > 0 {
```
Comment (akiioto) [inline cmd/image-builder/main.go:476]: I think this is not a mandatory check, because if source is empty then it will not execute at all

**Review by akiioto — APPROVED** [review]

**Pull request merged**